src/engraving/libmscore/tie.cpp

```diff
@@ -1141,11 +1143,16 @@ TieSegment* Tie::layoutFor(System* system)
         Chord* c1 = startNote()->chord();
         setTick(c1->tick());
         if (_slurDirection == DirectionV::AUTO) {
-            if (c1->measure()->hasVoices(c1->staffIdx(), c1->tick(), c1->actualTicks())) {
-                // in polyphonic passage, ties go on the stem side
-                _up = c1->up();
+            bool simpleException = st && st->isSimpleTabStaff();
+            if (st && st->isSimpleTabStaff()) {
+                _up = !(c1->voice() & 1);
             } else {
-                _up = !c1->up();
+                if (c1->measure()->hasVoices(c1->staffIdx(), c1->tick(), c1->actualTicks())) {
+                    // in polyphonic passage, ties go on the stem side
+                    _up = simpleException ? !(c1->voice() & 1) : c1->up();
```

Comment:
I have seen this code in many places, it is better to move it to some function with a clear name

`voice() & 1`

For example, you can add such functions to mscore.h:

```
inline constexpr bool isOddVoice(voice_idx_t voiceIdx) 
{
    return voiceIdx & 1;
}

inline constexpr bool isEvenVoice(voice_idx_t voiceIdx) 
{
    return !(voiceIdx & 1);
}
```

And use them here

Reply:
Done! I've only replaced the ones in tie.cpp. I've used the names `isUpVoice` and `isDownVoice` instead, because of the the code is 0-indexed but the display in-app is 1-indexed, which could create some unnecessary problems.